modules/docs/arrow-docs/docs/docs/patterns/typeclasspolymorphism/README.md

```diff
@@ -0,0 +1,906 @@
+---
+layout: docs
+title: Monad Comprehensions
+permalink: /docs/patterns/typeclass_polymorphism/
+---
+
+## TypeClass Polymorphism
+
+{:.advanced}
+advanced
+
+What if we could write apps without caring about the runtime data types used but just about __how the data is operated 
+on__? 
+
+Let's say we have an  application working with RxJava's `Observable`. We'll end up having a bunch of chained call stacks 
+based on that given data type. But at the end of the day, and for the sake of simplicity, wouldn't  `Observable` be just 
+like a "container" with some extra powers?
+
+And same story for other "containers" like `Flowable` , `Deferred` (coroutines), `Future`, `IO`, and many more.
```

Comment:
extra space before the first comma. Link integrations/rx2, integrations/kotlinxcoroutines, effects/io